### PR TITLE
feat(dop): Provide pipeline execution record api

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
Provide pipeline execution record api

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=316195&issueFilter__urlQuery=eyJzdGF0ZXMiOls0NDAyLDcxMDQsNzEwNSw0NDAzLDQ0MDQsNzEwNiw0NDA2LDQ0MDcsNDQxMiw0NTM4LDQ0MTMsNDQxNCw0NDE1LDQ0MTZdfQ%3D%3D&issueTable__urlQuery=eyJwYWdlTm8iOjEsICJwYWdlU2l6ZSI6MTB9&iterationID=1250&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    Provide pipeline execution record api          |
| 🇨🇳 中文    |    提供流水线执行记录接口          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkg/common/apis/context.go [hunk 1]
```diff
@@ -32,6 +32,12 @@ func WithInternalClientContext(ctx context.Context, internalClient string) conte
 	return transport.WithHeader(ctx, header)
 }
 
+func WithUserIDContext(ctx context.Context, userID string) context.Context {
+	header := transport.Header{}
+	header.Set("user-id", userID)
```
Comment (sfwn) [inline pkg/common/apis/context.go:37]: defines as `headerInternalClient`

**Review by sfwn — CHANGES_REQUESTED** [review]

File: api/proto/dop/projectpipeline/projectpipeline.proto [hunk 2]
```diff
@@ -360,4 +371,35 @@ message Pusher {
...
+message BatchCreateProjectPipelineResponse {}
+
+message ListPipelineExecHistoryRequest {
+    string name = 1;
+    repeated string executors = 2;
+    repeated string appNames = 3;
+    repeated string statuses = 4;
+    uint64 pageNo = 5;
+    uint64 pageSize = 6;
+    google.protobuf.Timestamp StartTimeBegin = 7;
```
[the remark this answers is not in the document]
Comment (littlejiancc) [reply api/proto/dop/projectpipeline/projectpipeline.proto]: done

File: pkg/common/apis/context.go (hunk 1 above)
Comment (littlejiancc) [reply pkg/common/apis/context.go]: done

**Review by sfwn — DISMISSED** [review]

**Pull request merged**